charts/replicated-library/values-example.yaml

```diff
@@ -352,3 +353,14 @@ persistence:
           requests:
             storage: 8Gi
         storageClassName: slow
+
+# -- Configure the support bundle for the chart here.
+# supportBundle can be enabled by setting supportBundle.enabled to true
+# By default the supportBundle will be enabled and the default spec will be installed
+# @default -- See below
+supportBundle:
+  # -- Enables or disables the support bundle
+  enabled: true
+  # -- Install Default Support Bundle Spec or not
+  # -- Default Spec: https://raw.githubusercontent.com/replicatedhq/troubleshoot-specs/main/in-cluster/default.yaml
+  installDefaultSpec: true
```

Comment:
I'd like to think through this API a bit real fast.

```
supportBundle:
  # -- Enables or disables the support bundle
  enabled: true
  # -- Install Default Support Bundle Spec or not
  # -- Default Spec: https://raw.githubusercontent.com/replicatedhq/troubleshoot-specs/main/in-cluster/default.yaml
  installDefaultSpec: true
```

To avoid just having spec field where people write their whole spec what about something like.

```
troubleshoot:
  preflights:
    # Preflight template stuff here
  supportBundles:
    replicated:
      enabled: false # off by default
      uri: https://our-in-cluster-default.yaml
    my-custom-bundle: # each bundle can be specified 
      enabled: true
      uri: # URI or any other support bundle API field configured here as well
      collectors:
        a: {} # include templates here per collector by collector name
        b: {}     
      analyzers:
        a: {} # include templates for analyzers here by analyzer name
```

So what this does is create a top level `troubleshoot` key for all things troubleshoot. I'm not sure if we want to do that or split out preflights and supportBundles each one as a top key, but that's what I'm looking for some discussion on.

I do want to know before we land what the final api is expected to look like so we don't have to change it once we go to add preflights.

Other things this proposal is doing differently, the `supportBundle` key is a map of maps, and each of the sub-maps is 1 support bundle. There are no configurations for default directly on support bundle. This makes it so that our default bundle and every other bundle a user writes are all configured the same way.

The other thing I'm showing here is how we could configure the default bundle using the same templates that we expect chart authors to use to write bundles. We could just include the `uri` field if we want, or we could actually configure `collectors` and `analyzer` directly in the chart in addition to our `uri` field so that anyone in air-gap environemnts get something even w/o internet access.

Reply:
Thank you for your advice. I have tested another branch. 
https://github.com/replicatedhq/helm-charts/commit/ce6ae142572be253863faca35f528666393ffecb#diff-f4248ed11f08f0b09cfe3fd614ad3f7fabf84956d4b318b9cc1e456223f177cb

It is quite nested helm templates to pass values. So I would suggest the API to be
```
supportBundles:
  replicated:
    enabled: true # off by default
    uri: https://raw.githubusercontent.com/replicatedhq/troubleshoot-specs/main/in-cluster/default.yaml
    collectors:
      - clusterInfo
  my-custom-bundle: # each bundle can be specified
    enabled: true
    collectors:
      - clusterInfo
      - kotsadm-postgres-db
```
Toubleshoot on top of supportBundles will make it more nested.

The current method didn't support collectors override, it is using the templates from our _collectors.tpl.

Another ways is 
```
  my-custom-bundle: # each bundle can be specified
    enabled: true
    collectors:
      - clusterInfo: {}
      - kotsadm-postgres-db:
        - exec: # this is removable when we don't need to support kots <= 1.87
            args:
              - "-U"
              - kotsadm
            collectorName: kotsadm-postgres-db
            command:
              - pg_dump
            containerName: kotsadm-postgres
            name: kots/admin_console
            selector:
              - app=kotsadm-postgres
            timeout: 10s
```

However, that will create a very long values.yaml.

Reply:
I think the latter of those two examples is appropriate, even if it does make a long values.yaml.  The reason is that we offer config on each collector, and don't have a better way to supply it.

Let's also bring the scope of this PR into perspective.  The discussion on the UX is important as a prereq to the first commit towards the feature, and if we're agreed on the following, then this PR needs to be set to match the first part (`supportBundles.replicated`):

```yaml
supportBundles:
  replicated:
    enabled: true # off by default
    uri: https://raw.githubusercontent.com/replicatedhq/troubleshoot-specs/main/in-cluster/default.yaml
  my-custom-bundle:  # each bundle can be specified
    enabled: true
      collectors:
      - clusterInfo: {}
      - collector2:
           configA: foo
```

Reply:
I would also like to keep preflights separate - they do share some code, but run in a different way at a different time for a different purpose.

Reply:
I'd like to dig in on that more. Look at how we handle Deployments, Daemonsets, and Statefulsets they're used for different things and we collapose those into a single `apps` key. Preflights and SupportBundles are effectively the same thing (excluding the Host version) and it seems duplicative to separate them.